internal/core/item_rendering.rs

```diff
@@ -531,19 +567,11 @@ impl<'a, T> PartialRenderer<'a, T> {
 
 macro_rules! forward_rendering_call {
     (fn $fn:ident($Ty:ty) $(-> $Ret:ty)?) => {
-        fn $fn(&mut self, obj: Pin<&$Ty>, item_rc: &ItemRc) $(-> $Ret)? {
+        fn $fn(&mut self, obj: Pin<&$Ty>, item_rc: &ItemRc, size: LogicalSize) $(-> $Ret)? {
             let mut ret = None;
             Self::do_rendering(&self.cache, &obj.cached_rendering_data, || {
-                ret = Some(self.actual_renderer.$fn(obj, item_rc));
-                type Ty = $Ty;
-                let width = Ty::FIELD_OFFSETS.width.apply_pin(obj).get_untracked();
-                let height = Ty::FIELD_OFFSETS.height.apply_pin(obj).get_untracked();
-                let x = Ty::FIELD_OFFSETS.x.apply_pin(obj).get_untracked();
-                let y = Ty::FIELD_OFFSETS.y.apply_pin(obj).get_untracked();
-                LogicalRect::new(
-                    LogicalPoint::from_lengths(x, y),
-                    LogicalSize::from_lengths(width, height),
-                )
+                ret = Some(self.actual_renderer.$fn(obj, item_rc, size));
+                obj.geometry()
```

Comment:
Here we track the geometry again so it can invalidate the partial rendering cache.  We do that here since the actual rendering do not query the geometry.  One optimization we could do in the future, is not do it like that and instead, compare with what's stored in the geometry cache.  (we anyway need to do a visit of all the item geometry to compute what region to draw)